oshell.sh

```diff
@@ -1,40 +1,69 @@
 #!/bin/zsh
+# shellcheck shell=bash disable=SC1071
 
-# Version: 0.1.0
+# Version: 0.1.1
 
 # Color definitions for terminal output
 CYAN='\033[0;96m'
 YELLOW='\033[;33m'
-UNSET_FMT=$(tput sgr0)
 RED='\033[0;31m'
 
-# Log file for the OCI authentication refresher
-REFRESHER_LOG_FILE="${HOME}/Library/Logs/oci-auth-refresher_${OCI_CLI_PROFILE}.log"
+if [[ -n "$TERM" ]]; then
+  UNSET_FMT=$(tput sgr0)
+else
+  UNSET_FMT='\033[0m'
+fi
+
+# Ensure OSHELL_HOME is exported so oci_auth_refresher.sh can source common functions
+export OSHELL_HOME="${OSHELL_HOME:-$(cd "$(dirname "${BASH_SOURCE[0]:-$0}")" && pwd)}"
```

Comment:
I don't see how the `cd` and `pwd` commands are getting utilized here.

Shouldn't this suffice?

```suggestion
export OSHELL_HOME="${OSHELL_HOME:-$(dirname "${BASH_SOURCE[0]:-$0}")}"
```

Reply:
the `cd ... && pwd` is there to turn relative paths into absolute ones. Using just `dirname` can return things like `.` or `./scripts`, depending on how the script is invoked. That can break later references to `$OSHELL_HOME/oshell.sh`.  The `cd && pwd` ensures we always get a full path like `/Users/me/project/oshell`, which makes sourcing or referencing sibling files more reliable. what you suggestion is not valid solution `export OSHELL_HOME="${OSHELL_HOME:-$(dirname "${BASH_SOURCE[0]:-$0}")}"`